applications/StructuralMechanicsApplication/custom_elements/timoshenko_beam_element_2D2N.cpp

```diff
@@ -0,0 +1,1051 @@
+        global_size_N.clear();
+        global_size_N[1] = N_shape[0];
+        global_size_N[2] = N_shape[1];
+        global_size_N[4] = N_shape[2];
+        global_size_N[5] = N_shape[3];
+        noalias(rRHS) += global_size_N * local_body_forces[1] * jacobian_weight * GetProperties()[CROSS_AREA];
+    }
+
+    RotateAll(rLHS, rRHS, r_geometry);
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void TimoshenkoBeamElement2D2N::CalculateLeftHandSide(
+    MatrixType& rLHS,
+    const ProcessInfo& rProcessInfo
+    )
+{
+    KRATOS_TRY;
+    const auto &r_geometry = GetGeometry();
+    const SizeType number_of_nodes = r_geometry.size();
+    const SizeType mat_size = GetDoFsPerNode() * number_of_nodes;
+
+    if (rLHS.size1() != mat_size || rLHS.size2() != mat_size) {
+        rLHS.resize(mat_size, mat_size, false);
+    }
+    noalias(rLHS) = ZeroMatrix(mat_size, mat_size);
+
+    const auto& integration_points = IntegrationPoints(GetIntegrationMethod());
+
+    ConstitutiveLaw::Parameters cl_values(r_geometry, GetProperties(), rProcessInfo);
+    auto &r_cl_options = cl_values.GetOptions();
+    r_cl_options.Set(ConstitutiveLaw::COMPUTE_STRESS             , true);
+    r_cl_options.Set(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR, true);
+
+    const double Phi    = CalculatePhi(cl_values);
+    const double length = StructuralMechanicsElementUtilities::CalculateReferenceLength2D2N(*this);
+    const double J      = 0.5 * length;
+
+    // Let's initialize the cl values
+    VectorType strain_vector(3), stress_vector(3);
+    MatrixType constitutive_matrix(3, 3);
+    strain_vector.clear();
+    cl_values.SetStrainVector(strain_vector);
+    cl_values.SetStressVector(stress_vector);
+    cl_values.SetConstitutiveMatrix(constitutive_matrix);
+    VectorType nodal_values(6);
+    GetNodalValuesVector(nodal_values);
+    VectorType global_size_N(6), N_u_derivatives(2), N_theta_derivatives(4), N_theta(4), N_derivatives(4);
+
+    // Loop over the integration points
+    for (SizeType IP = 0; IP < integration_points.size(); ++IP) {
+        global_size_N.clear();
+        const double xi     = integration_points[IP].X();
+        const double weight = integration_points[IP].Weight();
+        const double jacobian_weight = weight * J;
+
+        strain_vector[0] = CalculateAxialStrain(length, Phi, xi, nodal_values);      // El
+        strain_vector[1] = CalculateBendingCurvature(length, Phi, xi, nodal_values); // Kappa
+        strain_vector[2] = CalculateShearStrain(length, Phi, xi, nodal_values);      // Gamma_xy
+
+        mConstitutiveLawVector[IP]->CalculateMaterialResponseCauchy(cl_values);
+        const MatrixType& r_constitutive_matrix = cl_values.GetConstitutiveMatrix();
+        const double dN_dEl    = r_constitutive_matrix(0, 0);
```

Comment:
i would make this as a product. This is important in the cases in which the rConstitutiveMatrix is not diagonal

Reply:
this was not changed??